app/models/chat_channel.rb

```diff
@@ -27,6 +27,34 @@ class ChatChannel < ActiveRecord::Base
 
   validates :name, length: { maximum:  Proc.new { SiteSetting.max_topic_title_length } }, presence: true, allow_nil: true
 
+  def join(user)
```

Comment:
I you move this logic to `models/chat_channel.rb` there should be specs for this now